.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,26 @@
+name: github pages
+
+on:
+  push:
+    branches:
+    - documentation
```

Comment:
Ah, good catch. This is left over from how I was testing things. The documentation build should just be run when pushing to the master branch (unless we decide otherwise).
How I think things should work is;
- documentation markdown files live in the normal repository, in the docs directory.
- when anything is merged to master documentation markdown files get compiled by mdbook into html which is pushed to the gh-pages branch.
- gh-pages is a special branch name that is picked up by github pages. Any html there is published (in our case to https://xi-editor.github.io/druid)

I tested if this still works if there is anything on https://xi-editor.github.io and this did work on my own repository (I have stuff published on mendelt.github.io and mendelt.github.io/cmdr)

One downside to this way of publishing is that documentation is always published from the head of the master branch. It might be nicer to have documentation for the current published crates.io version there. Maybe we can figure out a better branching strategy in the future?

Reply:
okay, so if I merge this, what else needs to happen for this to work? Do I need to set up some key?

Reply:
It needs secrets.ACTIONS_DEPLOY_KEY set to something that allows it to publish to the gh-pages branch. I assume it will work then but didn't get around to testing publishing with github actions. I'm at Agadir airport right now waiting for my flight home. I was planning on playing with it tomorrow. But if you feel adventurous then go ahead :-)

Reply:
And you might need to configure github pages to pick up the branch. But I think github does that automagically as soon as you push something to gh-pages.

Reply:
I succesfully got this to work on my own repository by following the steps under "(1) Add SSH deploy key" here https://github.com/peaceiris/actions-gh-pages
I tried this on my own repository on https://github.com/mendelt/cmdr with exactly the same github action. So I'm pretty sure that once you add the secret in the repository settings this will just work after merging!

Reply:
great, I'll look into getting this merged & working tomorrow. :)